SU2_CFD/include/solvers/CNEMOEulerSolver.hpp

```diff
@@ -0,0 +1,912 @@
+  Total_HeatFluxDiff,   /*!< \brief Total Equivalent Area coefficient for all the boundaries. */
+  Total_MassFlowRate;   /*!< \brief Total Mass Flow Rate on monitored boundaries. */
+  su2double
+  *Surface_CL,         /*!< \brief Lift coefficient for each monitoring surface. */
+  *Surface_CD,         /*!< \brief Drag coefficient for each monitoring surface. */
+  *Surface_CSF,        /*!< \brief Side-force coefficient for each monitoring surface. */
+  *Surface_CEff,       /*!< \brief Side-force coefficient for each monitoring surface. */
+  *Surface_CFx,        /*!< \brief x Force coefficient for each monitoring surface. */
+  *Surface_CFy,        /*!< \brief y Force coefficient for each monitoring surface. */
+  *Surface_CFz,        /*!< \brief z Force coefficient for each monitoring surface. */
+  *Surface_CMx,        /*!< \brief x Moment coefficient for each monitoring surface. */
+  *Surface_CMy,        /*!< \brief y Moment coefficient for each monitoring surface. */
+  *Surface_CMz,        /*!< \brief z Moment coefficient for each monitoring surface. */
+  *Surface_HF_Visc,    /*!< \brief Total (integrated) heat flux for each monitored surface. */
+  *Surface_MaxHF_Visc; /*!< \brief Maximum heat flux for each monitored surface. */
+
+  su2double *iPoint_UndLapl, /*!< \brief Auxiliary variable for the undivided Laplacians. */
+  *jPoint_UndLapl;           /*!< \brief Auxiliary variable for the undivided Laplacians. */
+  su2double *SecondaryVar_i, /*!< \brief Auxiliary vector for storing the solution at point i. */
+  *SecondaryVar_j;           /*!< \brief Auxiliary vector for storing the solution at point j. */
+  su2double *PrimVar_i,      /*!< \brief Auxiliary vector for storing the solution at point i. */
+  *PrimVar_j;                /*!< \brief Auxiliary vector for storing the solution at point j. */
+  su2double
+  **LowMach_Precontioner; /*!< \brief Auxiliary vector for storing the inverse of Roe-turkel preconditioner. */
+  unsigned long nMarker,  /*!< \brief Total number of markers using the grid information. */
+  *nVertex;               /*!< \brief Store nVertex at each marker for deallocation */
+  bool space_centered,    /*!< \brief True if space centered scheeme used. */
+  euler_implicit,         /*!< \brief True if euler implicit scheme used. */
+  least_squares;          /*!< \brief True if computing gradients by least squares. */
+
+  su2double *Primitive,   /*!< \brief Auxiliary nPrimVar vector. */
+  *Primitive_i,           /*!< \brief Auxiliary nPrimVar vector for storing the primitive at point i. */
+  *Primitive_j;           /*!< \brief Auxiliary nPrimVar vector for storing the primitive at point j. */
+
+  su2double *Secondary,   /*!< \brief Auxiliary nPrimVar vector. */
+  *Secondary_i,           /*!< \brief Auxiliary nPrimVar vector for storing the primitive at point i. */
+  *Secondary_j;           /*!< \brief Auxiliary nPrimVar vector for storing the primitive at point j. */
```

Comment:
Please avoid using this kind of auxiliary variable, have a look at what is done in CEulerSolver currently.